qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -434,17 +435,11 @@ def inverse_2q(num: Integral) -> Integral:
 
 def num_from_2q_circuit(qc: QuantumCircuit) -> Integral:
     """Convert a given 2-qubit Clifford circuit to the corresponding integer."""
-    return _compose_num_with_circuit_2q(0, qc)
-
-
-def _compose_num_with_circuit_2q(num: Integral, qc: QuantumCircuit) -> Integral:
-    """Compose a number that represents a Clifford, with a Clifford circuit, and return the
-    number that represents the resulting Clifford."""
-    lhs = num
+    lhs = 0
     for inst in qc:
         qubits = tuple(qc.find_bit(q).index for q in inst.qubits)
         rhs = _num_from_2q_gate(op=inst.operation, qubits=qubits)
-        lhs = _CLIFFORD_COMPOSE_2Q[lhs, rhs]
+        lhs = _CLIFFORD_COMPOSE_2Q_DENSE[lhs, _clifford_num_to_dense_index[rhs]]
```

Comment:
```suggestion
    for rhs in _clifford_2q_nums_from_2q_circuit(qc):
        lhs = _CLIFFORD_COMPOSE_2Q[lhs, rhs]
```
It would be nice to use `_clifford_2q_nums_from_2q_circuit` here too. And you cannot use `_CLIFFORD_COMPOSE_2Q_DENSE` here since `qc` may not be a circuit defined in `_CLIFFORD_LAYER`.

Reply:
You are right. But it would be nice if we could make `_clifford_num_to_dense_index` contain all the valid entries from `_CLIFFORD_COMPOSE_2Q`, because then we can use the dense table. Your suggestion works because `_clifford_2q_nums_from_2q_circuit` apparently only returns indices which are valid for `_CLIFFORD_COMPOSE_2Q`, but that is not explicit.

How was `_CLIFFORD_COMPOSE_2Q` generated? And in particular: which entries are valid and which ones one.

Another option would be to use
```
   for rhs in _clifford_2q_nums_from_2q_circuit(qc):
        lhs = compose_2q(lhs, rhs)
```

Reply:
@itoko I checked which indices in de sparse table are valid with
```
w=_CLIFFORD_COMPOSE_2Q.T
valid_sparse_indices = [num for num in range(11520) if w[num,:].nnz==11520-1]
```
I put the indexing into the sparse table back once more. Let me know if you think this is incorrect.